src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt

```diff
@@ -36,10 +36,9 @@ class PersistentMenuHelper {
      fun advanceCurrentMenuWithInput(screen: Screen, input: String) {
         if (screen is MenuScreen) {
             val index = input.toInt()
-            if (currentMenu != null && index < currentMenu!!.commands.size) {
-                currentMenu =  currentMenu!!.commands[index]
-            } else if (index < persistentMenu!!.size) {
-                currentMenu =  persistentMenu!![index]
+            val parentMenu = if (currentMenu == null) persistentMenu else currentMenu!!.commands
+            if (index < parentMenu.size) {
```

Comment:
Would we ever expect index to be > parentMenu.size?

Reply:
not anymore since we are not handling for smart links to hidden menus, removed here - https://github.com/dimagi/formplayer/pull/1594/commits/cee21e73ca54d536d6011f0273174f75383c3425